lib/core/filtering/simple_rule.rb

```diff
@@ -34,26 +36,50 @@ class Rule
         GREATER_THAN = '>'
       end
 
-      attr_reader :policy, :field, :rule, :value, :id
+      attr_reader :policy, :field, :rule, :value, :id, :order
 
       def initialize(rule_hash)
-        @policy = rule_hash.fetch(POLICY)
-        @field = rule_hash.fetch(FIELD)
-        @rule = rule_hash.fetch(RULE)
-        @value = rule_hash.fetch(VALUE)
-        @id = rule_hash.fetch(ID)
+        @policy = SimpleRule.flex_fetch(rule_hash, POLICY)
+        unless @policy == Policy::INCLUDE || @policy == Policy::EXCLUDE
+          raise "Invalid policy '#{policy}' for rule '#{rule_hash}'"
+        end
+        @field = SimpleRule.flex_fetch(rule_hash, FIELD)
+        @rule = SimpleRule.flex_fetch(rule_hash, RULE)
+        @value = SimpleRule.flex_fetch(rule_hash, VALUE)
+        @id = SimpleRule.flex_fetch(rule_hash, ID)
+        @order = SimpleRule.flex_fetch(rule_hash, ORDER, 0)
         @rule_hash = rule_hash
       rescue KeyError => e
-        raise "#{e.key} is required"
+        raise "#{e.key} is required: #{e.message}"
+      end
+
+      def self.flex_fetch(hash, key, default_value = nil)
```

Comment:
I believe @vidok [said this will kill kittens](https://elastic.slack.com/archives/C03GAPTQ2JF/p1668507180031719?thread_ts=1668448379.617669&cid=C03GAPTQ2JF)

Reply:
Yeah if we do it in a monkey patch. =) But here, we don't.
Frankly, I don't see a good way to do it with key types unknown. Moreover, even the 
`hash[key] || hash.fetch(key.to_sym, nil) || hash.fetch(key.to_s, nil)` won't work properly in the case where the key = boolean false (learned the hard way, therefore this ugly code).

Reply:
When resolving merge conflicts, I convinced myself we didn't need this. We can just rely on string keys.